codemodder/codemods/harden_pyyaml.py

```diff
@@ -1,53 +1,45 @@
-    METADATA = CodemodMetadata(
-        DESCRIPTION=("Ensures all calls to yaml.load use `SafeLoader`."),
-        NAME="harden-pyyaml",
-        REVIEW_GUIDANCE=ReviewGuidance.MERGE_WITHOUT_REVIEW,
-    )
-    CHANGE_DESCRIPTION = "Ensures all calls to yaml.load use `SafeLoader`."
-    YAML_FILES = [
-        "harden-pyyaml.yaml",
-    ]
-
-    CHANGE_DESCRIPTION = "Adds `yaml.SafeLoader` to yaml.load calls"
-
-    def __init__(self, codemod_context: CodemodContext, file_context: FileContext):
-        SemgrepCodemod.__init__(self, file_context)
-        BaseTransformer.__init__(
-            self,
-            codemod_context,
-            self._results,
-            file_context.line_exclude,
-            file_context.line_include,
-        )
-
-    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
-        pos_to_match = self.node_position(original_node)
-        if self.filter_by_result(
-            pos_to_match
-        ) and self.filter_by_path_includes_or_excludes(pos_to_match):
-            line_number = pos_to_match.start.line
-            self.CHANGES_IN_FILE.append(
-                Change(str(line_number), self.CHANGE_DESCRIPTION).to_json()
-            )
-            second_arg = cst.Arg(value=cst.parse_expression("yaml.SafeLoader"))
-            return updated_node.with_changes(args=[*updated_node.args[:1], second_arg])
-        return updated_node
+from codemodder.codemods.base_codemod import ReviewGuidance
+from codemodder.codemods.api import SemgrepCodemod
+
+
+class HardenPyyaml(SemgrepCodemod):
+    NAME = "harden-pyyaml"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    DESCRIPTION = "Ensures all calls to yaml.load use `SafeLoader`."
+
+    @classmethod
+    def rule(cls):
+        return """
+            rules:
+              - id: harden-pyyaml
+                pattern-either:
+                  - patterns:
+                      - pattern: yaml.load(...)
+                      - pattern-inside: |
+                          import yaml
+                          ...
+                          yaml.load(...,$ARG)
+                      - metavariable-pattern:
+                          metavariable: $ARG
+                          patterns:
+                            - pattern-not:
+                                pattern: yaml.SafeLoader
+                  - patterns:
+                      - pattern: yaml.load(...)
+                      - pattern-inside: |
+                          import yaml
+                          ...
+                          yaml.load(...,Loader=$ARG)
+                      - metavariable-pattern:
+                          metavariable: $ARG
+                          patterns:
+                            - pattern-not:
+                                pattern: yaml.SafeLoader
+
+        """
+
+    def on_result_found(self, _, updated_node):
+        second_arg = cst.parse_expression("yaml.SafeLoader")
```

Comment:
Just a small thing but I think we should hide this behind a helper so that the user doesn't have to make reference to `libCST` if it's not necessary. It could maybe just become `self.parse_expression`.